src/Symfony/Component/Validator/Constraints/IbanValidator.php

```diff
@@ -64,51 +143,73 @@ public function validate($value, Constraint $constraint)
                 $this->context->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
                     ->addViolation();
             } else {
                 $this->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
+                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
                     ->addViolation();
             }
 
             return;
         }
 
-        // ...contain only digits and characters...
-        if (!ctype_alnum($canonicalized)) {
+        // ...contain uppercase characters only
+        if ($canonicalized !== strtoupper($canonicalized)) {
             if ($this->context instanceof ExecutionContextInterface) {
                 $this->context->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->setCode(Iban::INVALID_CASE_ERROR)
                     ->addViolation();
             } else {
                 $this->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CHARACTERS_ERROR)
+                    ->setCode(Iban::INVALID_CASE_ERROR)
                     ->addViolation();
             }
 
             return;
         }
 
-        // ...and contain uppercase characters only
-        if ($canonicalized !== strtoupper($canonicalized)) {
+        // ...start with a valid two-letter country code...
+        $countryCode = substr($canonicalized, 0, 2);
+        $countries = Intl::getRegionBundle()->getCountryNames();
+
+        if (!ctype_alpha($countryCode) || !isset($countries[$countryCode])) {
             if ($this->context instanceof ExecutionContextInterface) {
                 $this->context->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CASE_ERROR)
+                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
                     ->addViolation();
             } else {
                 $this->buildViolation($constraint->message)
                     ->setParameter('{{ value }}', $this->formatValue($value))
-                    ->setCode(Iban::INVALID_CASE_ERROR)
+                    ->setCode(Iban::INVALID_COUNTRY_CODE_ERROR)
+                    ->addViolation();
+            }
+
+            return;
+        }
+
+        // ...and have a valid format, if available
+        if (isset($this->countryFormats[$countryCode]) &&
+            false == preg_match('/^'.$this->countryFormats[$countryCode].'$/', $canonicalized)
```

Comment:
If preg_match returns false this is because because the regex is invalid in which case PHP will also give an E_WARNING (and thus the application crashes).

Symfony prefers to use strict comparisons, and as the warning is not suppressed, using a strict comparison is better then casting integer to boolean. :)